### DISCUSSION TITLE
What would happen when weed volume filled

### DISCUSSION THREAD

**Discussion by hamidreza-hosseini** [seaweedfs/seaweedfs#2406] [Q&A] [answered]
Hi @chrislusf ,
I've run a cluster of weed and assign some collection in it,
I think that for each collection assign request it creates 6 volume in my weed cluster,
Now after a year these volumes are filled ,
1.
How many volume weed would create for each volume id request for existing collection if all volumes have filled already ? will it create one volume for that collection untill it fill again and I request again or it has a default number for example 6 and it will create this number?
2.
If this number is 1 , how can I define and increase default number to more than one e.g: 10 to increase concurrency?
(I dont want to use grow api)
3.
Can I tell weed check number of free volumes automatically and if it is under specific number create some?

**Comment by chrislusf** [comment, marked as answer]
1. Create 6 new volumes when existing volumes are at 90% full.
2. N/A
3. Not clear what is the meaning of creating free volumes.

**Reply by hamidreza-hosseini** [reply]
>     1. Create 6 new volumes when existing volumes are at 90% full.

@chrislusf  Thanks.
1. are these numbers configurable? I mean both values (6 volumes and 90%)
And how?

**Reply by chrislusf** [reply]
`6 new volumes` can be configured in master.toml, generated by `weed scaffold -config=master`

90% is not configurable now.

**Reply by hamidreza-hosseini** [reply]
> `6 new volumes` can be configured in master.toml, generated by `weed scaffold -config=master`

1. Like following source?
`https://github.com/chrislusf/seaweedfs/wiki/Optimization#increase-concurrent-writes`

> `6 new volumes` can be configured in

2. Where I should put this toml file? in `-mdir=/etc/weedfs/master` directory of `all` weed masters and don't need to say anything about it to weed when it wants to run ?

**Reply by chrislusf** [reply]
1. yes
2. See the head of generated master.toml file.

**Reply by hamidreza-hosseini** [reply]
>     2. See the head of generated master.toml file.
@chrislusf 
Is the following path correct? I should put it in `/etc/seaweedfs/master.toml`?
I don't have this directory even ...
my master config directory that I've adjusted when I wanted to run weed is : `-mdir=/etc/weedfs/master`

```
root@master01:~# weed scaffold -config=master
# Put this file to one of the location, with descending priority
#    ./master.toml
#    $HOME/.seaweedfs/master.toml
#    /etc/seaweedfs/master.toml
# this file is read by master
```

**Comment by chrislusf** [comment]
Correct
